File: typescript/examples/transfers.ts
#!/usr/bin/env npx ts-node
// @ts-nocheck
/**
 * Transfers Example
 *
 * Transfer USD and spot assets between accounts and wallets.
 *
 * Requires: PRIVATE_KEY environment variable
 */

import { HyperliquidSDK } from '@quicknode/hyperliquid-sdk';

const PRIVATE_KEY = process.env.PRIVATE_KEY;

if (!PRIVATE_KEY) {
  console.error("Set PRIVATE_KEY environment variable");
  console.error("Example: export PRIVATE_KEY='0x...'");
  process.exit(1);
}

async function main() {
  const sdk = new HyperliquidSDK(undefined, { privateKey: PRIVATE_KEY });
  console.log(`Wallet: ${sdk.address}`);

  // Transfer USD to another address
  // const result = await sdk.transferUsd("0x1234567890123456789012345678901234567890", 10.0);
  // console.log(`USD transfer: ${JSON.stringify(result)}`);

  // Transfer spot asset to another address
  // const result = await sdk.transferSpot("PURR", "0x1234567890123456789012345678901234567890", 100.0);
  // console.log(`Spot transfer: ${JSON.stringify(result)}`);

  // Transfer from spot wallet to perp wallet (internal)
  // const result = await sdk.transferSpotToPerp(100.0);
  // console.log(`Spot to perp: ${JSON.stringify(result)}`);

  // Transfer from perp wallet to spot wallet (internal)
  // const result = await sdk.transferPerpToSpot(100.0);
  // console.log(`Perp to spot: ${JSON.stringify(result)}`);

  // Send asset (generalized transfer)
  // const result = await sdk.sendAsset("USDC", "100.0", "0x1234567890123456789012345678901234567890");
  // console.log(`Send asset: ${JSON.stringify(result)}`);

  console.log("Transfer methods available:");
  console.log("  sdk.transferUsd(destination, amount)");
  console.log("  sdk.transferSpot(token, destination, amount)");
  console.log("  sdk.transferSpotToPerp(amount)");
  console.log("  sdk.transferPerpToSpot(amount)");
  console.log("  sdk.sendAsset(token, amount, destination)");
}

main().catch(console.error);

File: python/examples/transfers.py
#!/usr/bin/env python3
"""
Transfers Example

Transfer USD and spot assets between accounts and wallets.

Requirements:
    pip install hyperliquid-sdk

Usage:
    export PRIVATE_KEY="0x..."
    python transfers.py
"""

import os
import sys

from hyperliquid_sdk import HyperliquidSDK

PRIVATE_KEY = os.environ.get("PRIVATE_KEY")

if not PRIVATE_KEY:
    print("Error: Set PRIVATE_KEY environment variable")
    print("  export PRIVATE_KEY='0xYourPrivateKey'")
    sys.exit(1)


def main():
    print("Hyperliquid Transfers Example")
    print("=" * 50)

    sdk = HyperliquidSDK(private_key=PRIVATE_KEY)
    print(f"Wallet: {sdk.address}")

    # Transfer USD to another address
    # result = sdk.transfer_usd("0x1234567890123456789012345678901234567890", 10.0)
    # print(f"USD transfer: {result}")

    # Transfer spot asset to another address
    # result = sdk.transfer_spot("PURR", "0x1234567890123456789012345678901234567890", 100.0)
    # print(f"Spot transfer: {result}")

    # Transfer from spot wallet to perp wallet (internal)
    # result = sdk.transfer_spot_to_perp(100.0)
    # print(f"Spot to perp: {result}")

    # Transfer from perp wallet to spot wallet (internal)
    # result = sdk.transfer_perp_to_spot(100.0)
    # print(f"Perp to spot: {result}")

    # Send asset (generalized transfer)
    # result = sdk.send_asset("USDC", "100.0", "0x1234567890123456789012345678901234567890")
    # print(f"Send asset: {result}")

    print()
    print("Transfer methods available:")
    print("  sdk.transfer_usd(destination, amount)")
    print("  sdk.transfer_spot(token, destination, amount)")
    print("  sdk.transfer_spot_to_perp(amount)")
    print("  sdk.transfer_perp_to_spot(amount)")
    print("  sdk.send_asset(token, amount, destination)")


if __name__ == "__main__":
    main()
